modin/experimental/core/io/sql/sql_dispatcher.py

```diff
@@ -0,0 +1,130 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""Module houses `SQLExperimentalDispatcher` class."""
+
+import warnings
+
+import pandas
+import numpy as np
+
+from modin.core.io import SQLDispatcher
+from modin.config import NPartitions
+
+
+class SQLExperimentalDispatcher(SQLDispatcher):
+    """Class handles experimental utils for reading SQL queries or database tables."""
+
+    __read_sql_with_offset = None
+
+    @classmethod
+    def preprocess_func(cls):  # noqa: RT01
+        """Prepare a function for transmission to remote workers."""
+        if cls.__read_sql_with_offset is None:
+            from modin.experimental.core.io.sql.utils import read_sql_with_offset
```

Comment:
Why is this not at the header?

Reply:
sql deposits are optional, so we only import if necessary.